client/my-sites/plan-features-2023-grid/components/storage-add-on-dropdown.tsx

```diff
@@ -52,57 +63,74 @@ export const StorageAddOnDropdown = ( {
 	label = '',
 	planSlug,
 	storageOptions,
-	showPrice = false,
 	onStorageAddOnClick,
+	priceOnSeparateLine = false,
 }: StorageAddOnDropdownProps ) => {
+	const translate = useTranslate();
 	const { gridPlansIndex } = usePlansGridContext();
-	const { storageAddOnsForPlan } = gridPlansIndex[ planSlug ];
+	const {
+		pricing: { currencyCode },
+		storageAddOnsForPlan,
+	} = gridPlansIndex[ planSlug ];
 	const { setSelectedStorageOptionForPlan } = useDispatch( WpcomPlansUI.store );
+	const isLargeCurrency = useIsLargeAddOnCurrency( {
+		storageAddOns: storageAddOnsForPlan,
+		currencyCode: currencyCode || 'USD',
+	} );
 	const selectedStorageOptionForPlan = useSelect(
-		( select ) => {
-			return select( WpcomPlansUI.store ).getSelectedStorageOptionForPlan( planSlug );
-		},
+		( select ) => select( WpcomPlansUI.store ).getSelectedStorageOptionForPlan( planSlug ),
 		[ planSlug ]
 	);
-
-	// TODO: Consider transforming storageOptions outside of this component
 	const selectControlOptions = storageOptions.map( ( storageOption ) => {
 		const title = getStorageStringFromFeature( storageOption.slug ) || '';
 		const price = getStorageOptionPrice( storageAddOnsForPlan, storageOption.slug );
 		return {
-			key: storageOption?.slug,
-			name: <StorageAddOnOption title={ title } price={ price } showPrice={ showPrice } />,
+			key: storageOption.slug,
+			name: <StorageAddOnOption title={ title } price={ price } />,
 		};
 	} );
 
-	const defaultStorageOption = storageOptions.find( ( storageOption ) => ! storageOption?.isAddOn );
-	const selectedOptionKey = selectedStorageOptionForPlan || defaultStorageOption?.slug || '';
-	const selectedOptionPrice = getStorageOptionPrice( storageAddOnsForPlan, selectedOptionKey );
-	const selectedOptionTitle = getStorageStringFromFeature( selectedOptionKey ) || '';
+	const selectedOptionKey = selectedStorageOptionForPlan
```

Comment:
**Note:**

Changes from l93-99 are cleaning up the empty string defaults for `selectedOptionKey` and `selectedOptionTitle`, but they aren't strictly necessary for this PR